apps/frontend/src/pages/projects/components/project-create-form/project-create-form.tsx

```diff
@@ -0,0 +1,68 @@
+import { Button, Input } from "~/libs/components/components.js";
+import { EMPTY_LENGTH } from "~/libs/constants/constants.js";
+import {
+	useAppForm,
+	useCallback,
+	useFormWatch,
+	useMemo,
+} from "~/libs/hooks/hooks.js";
+import {
+	type ProjectCreateRequestDto,
+	projectCreateValidationSchema,
+} from "~/modules/projects/projects.js";
+
+import { DEFAULT_PROJECT_CREATE_PAYLOAD } from "./libs/constants/constants.js";
+import styles from "./styles.module.css";
+
+type Properties = {
+	onSubmit: (payload: ProjectCreateRequestDto) => void;
+};
+
+const ProjectCreateForm = ({ onSubmit }: Properties): JSX.Element => {
+	const { control, errors, handleSubmit } = useAppForm<ProjectCreateRequestDto>(
+		{
+			defaultValues: DEFAULT_PROJECT_CREATE_PAYLOAD,
+			validationSchema: projectCreateValidationSchema,
+		},
+	);
+
+	const { name: enteredName } = useFormWatch({ control });
+
+	const isButtonDisabled = useMemo(
+		() => enteredName?.length === EMPTY_LENGTH,
+		[enteredName],
+	);
```

Comment:
I think we don't need this, we just use usual validation approach for all forms. First time validation triggers on submit (the button is not disabled) and then it won't submit until validation is passed. This is one of the common approaches - not to trigger validation until user tries to submit the form. And sometimes user doesn't know which fields are required or what is wrong with the values and why the button is disabled. 
```suggestion
```